lib/usd/ui/layerEditor/mayaSessionState.cpp

```diff
@@ -156,9 +156,23 @@ void MayaSessionState::unregisterNotifications()
 void MayaSessionState::mayaUsdStageReset(const MayaUsdProxyStageSetNotice& notice)
 {
     auto shapePath = notice.GetShapePath();
+    auto stage = notice.GetStage();
     if (shapePath == _currentProxyShapePath) {
```

Comment:
Main problem was that we only ever attempted to modify the proxy that was currently being displayed, but the Layer Editor does have UsdStage ptr's for all the proxy's in its dropdown list so we do need to update that.

I did initially try to reuse some existing signals that request the entire list be rebuilt, but I was finding that it wasn't a great experience.  By resetting everything, the Layer Editor gets emptied and the entire UI rebuilt which ends up being pretty annoying even if it doesn't take all that long.